### PR TITLE
feat(@desktop/wallet): Account view - New Account view UI

### LINKED ISSUE #11537: Account view - New Account view UI

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fixes #11537

### What does the PR do

new accounts view for wallet settings https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/%E2%9A%99%EF%B8%8F-Settings-%7C-Desktop?type=design&node-id=15479-316147&mode=dev

### Affected areas

Wallet Settings

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/status-im/status-desktop/assets/60327365/bbcfd062-f26a-4638-b722-fb0c71ca6788

### CODE REVIEW

File: src/app/modules/main/profile_section/keycard/module.nim [hunk 1]
```diff
@@ -213,7 +214,7 @@ proc addAccountsToKeycardItem(self: Module, item: KeycardItem, accounts: seq[Wal
     if acc.emoji.len == 0:
       icon = "wallet"
     item.addAccount(newKeyPairAccountItem(acc.name, acc.path, acc.address, acc.publicKey, acc.emoji, acc.colorId,
-      icon = icon, balance = 0.0))
+      icon = icon, balance = newCurrencyAmount()))
```
Comment (saledjenic) [inline src/app/modules/main/profile_section/keycard/module.nim:217]: @Khushboo-dev-cpp you could omit balance here.

File: src/app/modules/main/profile_section/keycard/module.nim [hunk 2]
```diff
@@ -271,7 +272,7 @@ proc buildDetailsViewKeycardItem(self: Module, keycard: KeycardDto): KeycardItem
   for ua in unknownAccountsAddresses:
     i.inc
     let name = atc.KEYCARD_ACCOUNT_NAME_OF_UNKNOWN_WALLET_ACCOUNT & $i
-    item.addAccount(newKeyPairAccountItem(name, path = "", ua, pubKey = "", emoji = "", colorId = "undefined", icon = "wallet", balance = 0.0))
+    item.addAccount(newKeyPairAccountItem(name, path = "", ua, pubKey = "", emoji = "", colorId = "undefined", icon = "wallet", balance = newCurrencyAmount()))
```
Comment (saledjenic) [inline src/app/modules/main/profile_section/keycard/module.nim:275]: @Khushboo-dev-cpp you could omit balance here.

File: storybook/pages/AccountViewPage.qml [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+
+//import AppLayouts.Wallet.panels 1.0
```
Comment (saledjenic) [inline storybook/pages/AccountViewPage.qml:4]: @Khushboo-dev-cpp leftover (lines below as well)?

File: ui/app/AppLayouts/Profile/views/wallet/AccountView.qml [hunk 4]
```diff
@@ -1,158 +1,193 @@
...
+            Separator {
+                Layout.fillWidth: true
+                Layout.preferredHeight: 1
+                color: Theme.palette.baseColor2
+                visible: derivationPath.visible
+            }
+            WalletAccountDetailsListItem {
+                Layout.fillWidth: true
+                title: qsTr("Stored")
+                subTitle: keyPair && keyPair.migratedToKeycard ? qsTr("on Keycard"): qsTr("On device")
```
Comment (saledjenic) [inline ui/app/AppLayouts/Profile/views/wallet/AccountView.qml:159]:
```suggestion
                subTitle: keyPair && keyPair.migratedToKeycard ? qsTr("On Keycard"): qsTr("On device")
```

File: ui/app/AppLayouts/Profile/views/wallet/AccountView.qml [hunk 5]
```diff
@@ -1,158 +1,193 @@
...
+                else
+                    console.warn("TODO: move keys to a Keycard")
+            }
+        }
+        StatusButton {
+            Layout.fillWidth: true
             objectName: "deleteAccountButton"
-            visible: !!root.account && root.account.walletType !== "" 
-            text: qsTr("Remove from your profile")
+            visible: !!root.account && root.account.walletType !== ""
```
Comment (saledjenic) [inline ui/app/AppLayouts/Profile/views/wallet/AccountView.qml:187]: `account.walletType` should never be `""`, so if you want to remove delete option for default wallet account you should check if `isWallet` is `true` instead (but ofc before that you should check if that role is exposed in this context, if not you will need to expose it).

**Comment by saledjenic** [comment]
I haven't tried locally but this is the screen shot from the attached video:



@Khushboo-dev-cpp @benjthayer now wondering do we need details view at all for watch only accounts? Or if we need it  then we need to adjust it a bit. 
1. `Balance` - ok
2. `Address` - ok, but it should not have prefixes (eth, opt, arb) since that's watch only account
3. `Keypair` - we should most likely remove this section, cause watch only accounts do not belong to any keypair
4. `Origin` - ok
5. `Derivation path` - we should most likely remove this section, cause we don't know derivation path for any watch only account
6. `Stored` - ok, but watch only accounts are always stored "on device"

**Comment by benjthayer** [comment]
Thanks @saledjenic !

> 3. `Keypair` - we should most likely remove this section, cause watch only accounts do not belong to any keypair
Agree we should remove this for watched addresses

> 5. `Derivation path` - we should most likely remove this section, cause we don't know derivation path for any watch only account
Yep this should not be included for watched addresses

> 6. `Stored` - ok, but watch only accounts are always stored "on device"
Yeah, probably best to keep in anyway to avoid any doubts and to be as consistent as possible in the presentation of account details across the different account types

> 7. `Migrate to a Keycard` - this option is not applicable at all for watch only accounts
No problem

I can add a watched address account detail screen to the wallet settings flows to clarify the additional restrictions / variation in presentation so we can reflect in the mobile flows as well.

**Comment by benjthayer** [comment]
> 2\. `Address` - ok, but it should not have prefixes (eth, opt, arb) since that's watch only account

Hi @John-44 could you confirm that we should remove network prefixes and the ability to alter network share preferences for watched addresses? Thanks!

**Comment by John-44** [comment]
> > 2. `Address` - ok, but it should not have prefixes (eth, opt, arb) since that's watch only account
> 
> Hi @John-44 could you confirm that we should remove network prefixes and the ability to alter network share preferences for watched addresses? Thanks!

I’m not sure actually, let’s discuss at next wallet design call

File: ui/app/AppLayouts/Profile/views/wallet/AccountView.qml (hunk 5 above)
Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Profile/views/wallet/AccountView.qml:187]: Aaah I dont know how I missed this, will do! I think this check should work 

walletType="generated" and path = "m/44'/60'/0'/0/0"
or
keyPair.type =Profile and path = "m/44'/60'/0'/0/0"

**Comment by Khushboo-dev-cpp** [comment]
> Thanks @saledjenic !
> 
> > 3. `Keypair` - we should most likely remove this section, cause watch only accounts do not belong to any keypair
> >    Agree we should remove this for watched addresses
> 
> > 5. `Derivation path` - we should most likely remove this section, cause we don't know derivation path for any watch only account
> >    Yep this should not be included for watched addresses
> 
> > 6. `Stored` - ok, but watch only accounts are always stored "on device"
> >    Yeah, probably best to keep in anyway to avoid any doubts and to be as consistent as possible in the presentation of account details across the different account types
> 
> > 7. `Migrate to a Keycard` - this option is not applicable at all for watch only accounts
> >    No problem
> 
> I can add a watched address account detail screen to the wallet settings flows to clarify the additional restrictions / variation in presentation so we can reflect in the mobile flows as well.

Hi @saledjenic , made it as per Bens new design he added for watch only accounts, thanks for pointing it out, looks like this now 
Screenshot 2023-07-19 at 1 25 16 PM

**Comment by Khushboo-dev-cpp** [comment]
> Thanks @saledjenic !
> 
> > 3. `Keypair` - we should most likely remove this section, cause watch only accounts do not belong to any keypair
> >    Agree we should remove this for watched addresses
> 
> > 5. `Derivation path` - we should most likely remove this section, cause we don't know derivation path for any watch only account
> >    Yep this should not be included for watched addresses
> 
> > 6. `Stored` - ok, but watch only accounts are always stored "on device"
> >    Yeah, probably best to keep in anyway to avoid any doubts and to be as consistent as possible in the presentation of account details across the different account types
> 
> > 7. `Migrate to a Keycard` - this option is not applicable at all for watch only accounts
> >    No problem
> 
> I can add a watched address account detail screen to the wallet settings flows to clarify the additional restrictions / variation in presentation so we can reflect in the mobile flows as well.

@saledjenic thanks for pointing it out, I have made it as per Bens design for watched account, in case there are changes next week, can accomodate it in a separate task

Screenshot 2023-07-19 at 1 46 16 PM

**Review by saledjenic — APPROVED** [review]

**Review by alaibe — APPROVED** [review]

**Comment by Khushboo-dev-cpp** [comment]
@anastasiyaig thanks for pointing out, the view looks like this now 

image

File: ui/imports/utils/Utils.qml [hunk 6]
```diff
@@ -786,6 +786,11 @@ QtObject {
         return Theme.palette.customisationColors.blue
     }
 
+    function getPathForDisplay(path) {
+        let letters = path.split("/").join(" / ")
```
Comment (saledjenic) [inline ui/imports/utils/Utils.qml:790]:
```suggestion
        return path.split("/").join(" / ")
```

**Review by anastasiyaig — APPROVED** [review]

**Pull request merged**